### PR TITLE
Finish first raking version

### LINKED ISSUE #94: Get color code map from scene renderer
We need to draw color codes to a buffer so it can be used by the rake renderer.

### LINKED ISSUE #98: Use color arrays for shader instead of multiple uniforms
Arrays in shader provide the advantage to shorten code on both side of the interface and make it more flexible

### LINKED ISSUE #103: Add enhanced direction support for rakes
Strokes from right to left (and down to up) should be possible.

### PR DESCRIPTION
**Pull request by t-erxleben**
closes #98 
closes #94
closes #103

### CODE REVIEW

File: backend/include/WGLRakeRenderer.hpp [hunk 1]
```diff
@@ -117,6 +111,7 @@ class WGLRakeRenderer: private WGLRenderer
 
         WGLRakeRenderer(WGLSceneRenderer& sr, Scene const & s);
         void reset(WGLSceneRenderer& sr, Scene const & s);
-        void rakeAndDraw(float x, float y, float speed, bool nails[100]);
+        void rake(float x, float y, float speed, bool nails[100]);
```
Comment (JBenda) [inline backend/include/WGLRakeRenderer.hpp]: Should also be changed to 1000

**Comment by t-erxleben** [comment]
I would suggest merging this and integrate it in the interface in another pr where we also introduce proper state switching (dropping vs. raking).

**Review by JBenda — COMMENTED** [review]
The back and for switching may be broken, but can you use the functions:
`startRaking` and `rakingLinear` with the code needed to work for without back switching.

Also, may add an example in main so than we can see how it's look (can be commended out, or in an `#ifdef`).

**Review by JBenda — APPROVED** [review]
Sorry, I have missed the example images :+1:

File: backend/include/WGLRakeRenderer.hpp (hunk 1 above)
Comment (t-erxleben) [reply backend/include/WGLRakeRenderer.hpp]: good catch :)

**Pull request merged**